### DISCUSSION TITLE
Idle disconnect

### DISCUSSION THREAD

**Discussion by yaroz** [rustdesk/rustdesk#5540] [Feature Request]
Good morning.  We have recently deployed the pro version of Rust Desk to our servers, and went to enable the "idle disconnect" like we have for Anydesk or TeamViewer, and it wasn't there.  This is an important security feature with servers, because if the client maintains an active connection to the server, even when idle, the server will not lock.

Basically, if we have a session open with the server and we switch to a different window, we would like it to disconnect after a certain time frame and lock the screen.  The time frames would be "never, 5, 10, 15, 30, 1h, 2h, 4h, 8h" or something like that.

**Comment by rustdesk** [comment]
Can you send me a link about "idle disconnect" for AnyDesk and TeamViewer?

**Comment by yaroz** [comment]
Sure.. in Anydesk, it's actually a setting called Auto-Disconnect.  The lock on exit would be nice as well.
![image](https://github.com/rustdesk/rustdesk/assets/20463573/ab7b76d2-4b77-48ca-9b4f-a75fa0b07ebd)

https://support.anydesk.com/knowledge/status-desk_rt_auto_disconnect discusses the setting

https://community.teamviewer.com/English/kb/articles/5614-time-out-inactive-sessions for teamviewer

**Comment by rustdesk** [comment]
@21pages do this. This is a simple feature. Easy to implement. Check default setting in AnyDesk and TeamViewer.

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/pull/5674
